crates/sparrow-main/src/serve/file_service.rs

```diff
@@ -105,13 +136,18 @@ pub(crate) async fn get_source_metadata(
         file_path::Path::CsvData(data) => file_path::Path::CsvData(data.to_string()),
     };
 
-    let metadata = RawMetadata::try_from(&source)?;
-    let schema = Schema::try_from(metadata.table_schema.as_ref()).with_context(|| {
-        format!(
-            "Unable to encode schema {:?} for source file {:?}",
-            metadata.table_schema, source
-        )
-    })?;
+    let metadata = RawMetadata::try_from(&source)
+        .into_report()
+        .change_context(Error::SchemaError("unable to get raw metadata".to_owned()))?;
```

Comment:
could `attach_printable_lazy!(format!("{}", schema)`, that would help debugging.

Reply:
We could also have one that makes it easier to print the schema -- although here, we don't yet have the schema I think...

Reply:
Added